superset-frontend/src/SqlLab/components/RunQueryActionButton.tsx

```diff
@@ -17,75 +17,114 @@
  * under the License.
  */
 import React from 'react';
-import { t } from '@superset-ui/core';
+import { t, styled, supersetTheme } from '@superset-ui/core';
 
-import { Dropdown, Menu } from 'src/common/components';
+import { Menu } from 'src/common/components';
+import Button, { ButtonProps } from 'src/components/Button';
 import Icon from 'src/components/Icon';
-
-const NO_OP = () => undefined;
+import { DropdownButton, DropdownProps } from 'src/common/components/Dropdown';
 
 interface Props {
   allowAsync: boolean;
-  dbId?: number;
   queryState?: string;
   runQuery: (c?: boolean) => void;
   selectedText?: string;
   stopQuery: () => void;
   sql: string;
-  overlayCreateAsMenu: typeof Menu;
+  overlayCreateAsMenu: typeof Menu | null;
 }
 
+type QueryButtonProps = DropdownProps | ButtonProps;
+
+const buildText = (
+  shouldShowStopButton: boolean,
+  selectedText: string | undefined,
+): string | JSX.Element => {
+  if (shouldShowStopButton) {
+    return (
+      <>
+        <i
+          className="fa fa-stop"
+          style={{ display: 'inline-block', marginRight: '10px' }}
```

Comment:
can we set this as var above the build text vs inline

Reply:
ah, thanks. I meant to come back to that.